tests/mentionui.js

```diff
@@ -560,6 +560,43 @@ describe( 'MentionUI', () => {
 					} );
 			} );
 
+			it( 'should not show panel when selection moves inside existing mention', () => {
+				setData( model, '<paragraph>foo [@Lily] bar</paragraph>' );
+
+				model.change( writer => {
+					writer.setAttribute( 'mention', { id: '@Lily', _uid: 1234 }, doc.selection.getFirstRange() );
+				} );
+
+				return waitForDebounce()
+					.then( () => {
+						expect( panelView.isVisible ).to.be.false;
+
+						model.change( writer => {
```

Comment:
This particular part doesn't bring any value to the tests as it puts the selection `foo @Lily ^bar`, which obviously should not trigger a selection but is tested elsewhere. So this step should be skipped.

The test should put following selections:
* `foo @Lily^ bar`
* `foo @Lil^y bar`

Reply:
Fixed in 7879d5c.

Reply:
I don'tr recall that case ATM but it might be issue with moving selection from right to left in that specificy scenario.

Reply:
It couldn't be anything like that, I have double checked it against `src/mentionui.js` `src/textwatcher.js` at (former) `master` branch and these assertions obviously didn't fail.

The simplified assertion does the job of detecting the former issue with less steps, and timeouts.